DataScience/Analytics/data/data.py

```diff
@@ -314,11 +324,101 @@ def convertCSVtoSQLite(path='/Users/saminiemi/Projects/ONS/AddressIndex/data/ADD
         df.to_sql('ab', cnx, index=False, if_exists='replace')
 
 
+def create_NLP_index(path='/Users/saminiemi/Projects/ONS/AddressIndex/data/ADDRESSBASE/', filename='NLPindex.csv'):
+    """
+    Read in all the Address Base Epoch 39 CSV files and combine to a single NLP index.
+
+    :param path: location of the AddressBase CSV files
+    :type path: str
+    :param filename: name of the output file
+    :type filename: str
+
+    :return: None
+    """
+    files = glob.glob(path + 'ABP_E39_*.csv')
+
+    columns = {'BLPU': ['UPRN', 'POSTCODE_LOCATOR'],
+               'LPI': ['UPRN', 'USRN', 'LANGUAGE', 'PAO_TEXT', 'PAO_START_NUMBER', 'PAO_START_SUFFIX', 'PAO_END_NUMBER',
+                       'PAO_END_SUFFIX','SAO_TEXT', 'SAO_START_NUMBER', 'SAO_START_SUFFIX', 'OFFICIAL_FLAG'],
+               'STREET_DESC': ['USRN', 'STREET_DESCRIPTOR', 'TOWN_NAME', 'LANGUAGE', 'LOCALITY'],
+               'ORGANISATION': ['UPRN', 'ORGANISATION']}
+
+    data_frames = {}
+    for file in tqdm(files):
+        if 'CLASSIFICATION' in file or 'STREET.csv' in file or 'DELIVERY_POINT' in file:
+            continue
```

Comment:
I agree. Any suggestions how to avoid this? I'd rather not these for all the files I need to use, but rather skip those that I won't need. I suppose I could always wrap the following in and if and advance to it only when the file name is not in the list of names to skip...